gulp/versioncheck.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var request = require('request');
+var pluginUtils = require('../lib/plugin-utils');
+var supportedMinVersion = 15.7;
+
+module.exports = function (gulp, gutil) {
+
+    function process(server, cb) {
+        try {
+            if (server.serverUrl() === undefined) {
+                var errMessage = 'A server URL is required in your configuration. ';
+                callbackOrThrowError(cb, errMessage);
+            }
+            var versionCheckUrl = pluginUtils.urlBldr(server.serverUrl() + 'status/version').build();
+            request(versionCheckUrl, function (error, response, body) {
+                if (error || response.statusCode > 201) {
+                    callbackOrThrowError(cb, error.message);
+                } else {
+                    var matches = body.match(/<body>(\d+(\.\d)*)/i);
+                    if (matches) {
+                        var versionOnServer = matches[1];
+                        if (versionOnServer < supportedMinVersion) {
+                            var errorMessage = "Supported minimum version on server is " + supportedMinVersion;
```

Comment:
Maybe we should say something extra like 'Either contact support to get your stage server upgraded to version' + version + 'or else downgrade your version of the sdk to ' + olderSdkVersion